src/open_inwoner/accounts/signals.py

```diff
@@ -26,6 +28,34 @@
 }
 
 
+@receiver(company_branch_selected)
+def update_company_from_kvk_api(sender, *args, **kwargs):
+    request = kwargs["request"]
+    user = request.user
+    branch_number = kwargs["vestigingsnummer"]
+
+    kvk_client = KvKClient()
+
+    vestiging = kvk_client.get_vestiging(vestigingsnummer=branch_number)
+
+    binnenlandsadres = vestiging.get("adres", {}).get("binnenlandsAdres", {})
+
+    fields = {
+        "company_name": vestiging.get("naam"),
+        "city": binnenlandsadres.get("plaats"),
+        "postcode": binnenlandsadres.get("postcode"),
+        "street": binnenlandsadres.get("straatnaam"),
+        "housenumber": binnenlandsadres.get("huisnummer"),
+    }
+
+    for field, new_value in fields.items():
+        current_value = getattr(user, field, None)
+        if new_value and new_value != current_value:
+            setattr(user, field, new_value)
```

Comment:
I wouldn't be enough to check for the presence of `binnenlandAdres`, because each field in `binnenlandAdres` is potentially null and would have to be checked as well. Some of the fields may actually be required (an address without at least city name would be strange), but I don't know which ones (the spec doesn't say). I find it simpler therefore to `get` the values from the `dict` and check if they are different from what's stored on the user.